### PR TITLE
Optimized the dependency metadata structure

### PR DESCRIPTION
**Pull request by pratikspatil024**
In this PR, I removed the first 2 elements in the TxDependency structure: the Transaction index and flag informing about delaying the fee transfer.

### CODE REVIEW

File: core/parallel_state_processor.go [hunk 1]
```diff
@@ -315,9 +315,11 @@ func (p *ParallelStateProcessor) Process(block *types.Block, statedb *state.Stat
 
 		cleansdb := statedb.Copy()
 
-		if len(header.TxDependency) > 0 {
-			shouldDelayFeeCal = delayMap[i]
+		if msg.From() == coinbase {
+			shouldDelayFeeCal = false
+		}
 
+		if len(header.TxDependency) > 0 {
```
Comment (cffls) [inline core/parallel_state_processor.go:322]: Should we check if `len(header.TxDependency) == len(tasks)`? If lengths mismatch, we shouldn't use any dependency data.

File: miner/worker.go [hunk 2]
```diff
@@ -1133,30 +1133,11 @@ func (w *worker) commitTransactions(env *environment, txs *types.TransactionsByP
...
-				tempDeps[i] = []uint64{uint64(i)}
-
-				reads := mvReadMapList[i-1]
-
-				_, ok1 := reads[blockstm.NewSubpathKey(env.coinbase, state.BalancePath)]
-				_, ok2 := reads[blockstm.NewSubpathKey(common.HexToAddress(w.chainConfig.Bor.CalculateBurntContract(env.header.Number.Uint64())), state.BalancePath)]
-
-				if ok1 || ok2 {
-					// 0 -> delay is not allowed
-					tempDeps[i] = append(tempDeps[i], 0)
```
Comment (cffls) [inline miner/worker.go:1154]: I think we should still keep this check. When delay is not allowed, we simply set dependency to empty, so validators will directly run serial execution.

File: core/parallel_state_processor.go (hunk 1 above)
Comment (pratikspatil024) [reply core/parallel_state_processor.go:322]: Nice idea. Thanks!

File: miner/worker.go (hunk 2 above)
Comment (pratikspatil024) [reply miner/worker.go:1154]: Yes, that will be nice.

**Review by cffls — APPROVED** [review]

**Pull request merged**